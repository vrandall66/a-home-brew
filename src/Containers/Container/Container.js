import React from 'react';
import BeerCard from '../BeerCard/BeerCard';
import './Container.scss';

export const Container = ({ beers, type, list }) => {
  // if list.length is true:
  // 

  // const beersToRender = list ? (reduce beers to only include beers on the list) : (beers)


  // CHANGE to iterating through 
  const beersLength = beers.length ? (
    // CHANGE BEERS TO beersToRender
    beers.map(beer => {
      return <BeerCard key={beer.id} beer={beer} type={type} />;
    })
  ) : (
    <div>{`There are no ${type}!`}</div>
  );

  return <div className='Container'>{beersLength}</div>;
};

export default Container;
