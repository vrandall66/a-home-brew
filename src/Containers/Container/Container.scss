.Container {
  display: grid;
  flex-wrap: wrap;
  grid-area: container;
  grid-gap: 15px 15px;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  grid-template-rows: auto;
  height: 66vh;
  justify-items: center;
  justify-content: space-between;
  margin-top: 8%;
  margin-bottom: 2%;
  overflow: scroll;
  padding: 0px 20px;
  width: 80vw;
}

a {
  margin: 5% 0%;
  text-decoration: none;
}

.Nav__h4--your-brewery {
  padding: 5% 0%;
}
