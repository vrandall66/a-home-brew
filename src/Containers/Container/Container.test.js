import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  let wrapper;
  let type = 'bookmarks';
  let beers = [
    { id: 1 },
    { id: 2 },
    { id: 3 },
    { id: 4 },
    { id: 5 },
    { id: 6 }
  ];
  let bookmarks = [1, 2];
  let previous = [4, 5, 6];

  beforeEach(() => {
    wrapper = shallow(
      <Container
        beers={beers}
        bookmarks={bookmarks}
        previous={previous}
        type={type}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should get bookmarked status', () => {
    let id = 2;
    const results = wrapper.getBookmarkStatus(id)

    expect(results).toEqual(true)
  });
});
