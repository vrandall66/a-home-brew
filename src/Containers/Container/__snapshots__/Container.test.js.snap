// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
<div
  className="Container"
  key="beers"
>
  <Connect(BeerCard)
    beer={
      Object {
        "id": 1,
      }
    }
    bookmark={true}
    key="1"
    previous={false}
    type="bookmarks"
  />
  <Connect(BeerCard)
    beer={
      Object {
        "id": 2,
      }
    }
    bookmark={true}
    key="2"
    previous={false}
    type="bookmarks"
  />
  <Connect(BeerCard)
    beer={
      Object {
        "id": 3,
      }
    }
    bookmark={false}
    key="3"
    previous={false}
    type="bookmarks"
  />
  <Connect(BeerCard)
    beer={
      Object {
        "id": 4,
      }
    }
    bookmark={false}
    key="4"
    previous={true}
    type="bookmarks"
  />
  <Connect(BeerCard)
    beer={
      Object {
        "id": 5,
      }
    }
    bookmark={false}
    key="5"
    previous={true}
    type="bookmarks"
  />
  <Connect(BeerCard)
    beer={
      Object {
        "id": 6,
      }
    }
    bookmark={false}
    key="6"
    previous={true}
    type="bookmarks"
  />
</div>
`;
